mypy/fastparse.py

```diff
@@ -271,7 +271,20 @@ def visit(self, node: Optional[AST]) -> Any:
             method = 'visit_' + node.__class__.__name__
             visitor = getattr(self, method)
             self.visitor_cache[typeobj] = visitor
-        return visitor(node)
+        result = visitor(node)
+        if (3, 8) < sys.version_info and isinstance(node, ast3.expr):
+            self.scope_ignores(node)
+        return result
+
+    def scope_ignores(self, node: ast3.expr) -> None:
+        end_lineno = getattr(node, "end_lineno", None)
+        if end_lineno is None:
+            return
+        node_lines = range(node.lineno, end_lineno + 1)
+        for line in node_lines:
+            if line in self.type_ignores.values():
```

Comment:
I am not sure how efficient is this. I think it would be better to assign `mod.type_ignores` to an attribute on `self` and use it for this check instead of recomputing this same set every time.

Also please add a comment with an example explaining what is going on here.